examples/platform/efr32/BaseApplication.cpp

```diff
@@ -0,0 +1,517 @@
+
+bool sIsProvisioned      = false;
+bool sIsEnabled          = false;
+bool sIsAttached         = false;
+bool sHaveBLEConnections = false;
+
+#endif // CHIP_DEVICE_CONFIG_ENABLE_SED
+
+EmberAfIdentifyEffectIdentifier sIdentifyEffect = EMBER_ZCL_IDENTIFY_EFFECT_IDENTIFIER_STOP_EFFECT;
+
+uint8_t sAppEventQueueBuffer[APP_EVENT_QUEUE_SIZE * sizeof(AppEvent)];
+StaticQueue_t sAppEventQueueStruct;
+
+StackType_t appStack[APP_TASK_STACK_SIZE / sizeof(StackType_t)];
+StaticTask_t appTaskStruct;
+
+BaseApplication::Function_t mFunction;
+bool mFunctionTimerActive;
+
+Identify * gIdentifyptr = nullptr;
+
+} // namespace
+
+/**********************************************************
+ * AppTask Definitions
+ *********************************************************/
+
+CHIP_ERROR BaseApplication::StartAppTask(TaskFunction_t taskFunction)
+{
+    sAppEventQueue = xQueueCreateStatic(APP_EVENT_QUEUE_SIZE, sizeof(AppEvent), sAppEventQueueBuffer, &sAppEventQueueStruct);
+    if (sAppEventQueue == NULL)
+    {
+        EFR32_LOG("Failed to allocate app event queue");
+        appError(APP_ERROR_EVENT_QUEUE_FAILED);
+    }
+
+    // Start App task.
+    sAppTaskHandle =
+        xTaskCreateStatic(taskFunction, APP_TASK_NAME, ArraySize(appStack), &sAppEventQueue, 1, appStack, &appTaskStruct);
+    if (sAppTaskHandle == nullptr)
+    {
+        EFR32_LOG("Failed to create app task");
+        appError(APP_ERROR_CREATE_TASK_FAILED);
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BaseApplication::Init(Identify * identifyObj)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (identifyObj == nullptr)
+    {
+        EFR32_LOG("funct timer create failed");
+        appError(CHIP_ERROR_INVALID_ARGUMENT);
+    }
+
+    gIdentifyptr = identifyObj;
+
+#ifdef SL_WIFI
+    /*
+     * Wait for the WiFi to be initialized
+     */
+    EFR32_LOG("APP: Wait WiFi Init");
+    while (!wfx_hw_ready())
+    {
+        vTaskDelay(10);
+    }
+    EFR32_LOG("APP: Done WiFi Init");
+    /* We will init server when we get IP */
+
+    sWiFiNetworkCommissioningInstance.Init();
+#endif
+
+    // Create FreeRTOS sw timer for Function Selection.
+    sFunctionTimer = xTimerCreate("FnTmr",                  // Just a text name, not used by the RTOS kernel
+                                  1,                        // == default timer period (mS)
+                                  false,                    // no timer reload (==one-shot)
+                                  (void *) this,            // init timer id = app task obj context
+                                  FunctionTimerEventHandler // timer callback handler
+    );
+    if (sFunctionTimer == NULL)
+    {
+        EFR32_LOG("funct timer create failed");
+        appError(APP_ERROR_CREATE_TIMER_FAILED);
+    }
+
+    // Create FreeRTOS sw timer for LED Management.
+    sLightTimer = xTimerCreate("LightTmr",            // Text Name
+                               10,                    // Default timer period (mS)
```

Comment:
this breaks the sleep/power consumption optimization done in the light switch-app

Reply:
After discussion on slack, we agreed to keep this since call to `StartLightTimer` and `StopLightTimer` are enabled for SED